### PR TITLE
setup.py: On Unix install scripts without .py suffix

### LINKED ISSUE #462: setup.py: Install scripts without extensions on UNIX-like platforms
Because of issue #394 , meson install scripts as `xxx.py` , but in linux, install script with a extension name is no a good practice. And change the installed script name also break some package depend on meson.

Could you deal with it for different platform?

thx.

### PR DESCRIPTION
**Pull request by TingPing**
I saw this was done previously but reverted because of Windows, so this just conditionally does it.
I reused most of the logic from distutils directly and we only use a subset of features so the behavior should be the same.

Closes #462

### CODE REVIEW

**Comment by jpakkane** [comment]
This may have wacky implications so let's merge it only after the next release (hopefully today/tomorrow).

File: setup.py [hunk 1]
```diff
@@ -25,8 +27,36 @@
...
+        # We want the files to be installed without a suffix on Unix
+        for infile in self.get_inputs():
+            in_stripped = infile[:-3] if infile.endswith('.py') else infile
+            outfile = path.join(self.install_dir, in_stripped)
+            copy_file(infile, outfile, dry_run=self.dry_run)
+            self.outfiles.append(outfile)
+
+        if not self.dry_run:
+            for f in self.get_outputs():
+                mode = ((os.stat(f)[ST_MODE]) | 0o555) & 0o7777
```
Comment (jpakkane) [inline setup.py]: Instead of setting flags blindly we should just preserve what the original files have. Or is this not possible with Distutils?

Comment (TingPing) [inline setup.py]: This was literally just a copy paste of the current behavior, I can look into `copy_file()` again and see if it gives us the option to preserve modes.

**Comment by TingPing** [comment]
@jpakkane I removed manually setting the mode, `copy_file()` preserves modes by default.

**Pull request merged**